### ISSUE TITLE
test_vsock fails on Linux under QEMU

### ISSUE THREAD

**Issue by asomers** [nix-rust/nix#1403]
Sometime in between 6-March-2021 and 16-March-2021, `sys::test_socket::test_vsock` has begin to fail for the QEMU architectures (but it still passes on x86_64 Linux).

For example, from https://github.com/nix-rust/nix/pull/1401/checks?check_run_id=2124256635 ,
```
---- sys::test_socket::test_vsock stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `None`,
 right: `Some(Sys(EADDRNOTAVAIL))`', test/sys/test_socket.rs:1517:5
```
@stefano-garzarella @qwandor-google as the only two known users of Vsock with Nix, can one of you please debug this?

**Comment by asomers** [comment]
Possibly the same root cause as #1310

**Comment by stefano-garzarella** [comment]
@asomers sure! I can take a look!

**Comment by stefano-garzarella** [comment]
The test is failing here:
```rust
    let sockaddr = SockAddr::new_vsock(libc::VMADDR_CID_LOCAL, port);
    assert_eq!(bind(s1, &sockaddr).err(),
               Some(Error::Sys(Errno::EADDRNOTAVAIL)));
```

Starting from Linux 5.6, `VMADDR_CID_LOCAL` is supported to do local communication (loopback device).
Before this it was reserved, but probably the test runner now has a newer kernel and `VMADDR_CID_LOCAL` is supported.

I'll send a patch right away to remove that check and re-enable the test.

**Comment by asomers** [comment]
Fixed by #1407 .
